### PR TITLE
Make GenericMap.plot() arguments keyword only

### PR DESCRIPTION
**Pull request by yeppAniket**
Made the suggested changes, added a changelog and a what's new section. for #7166

### CODE REVIEW

**Comment by nabobalis** [comment]
In the future you can update the same branch and only have to open one pull request.

**Comment by yeppAniket** [comment]
> In the future you can update the same branch and only have to open one pull request.

Oh okay

File: docs/whatsnew/6.0.rst [hunk 1]
```diff
@@ -44,3 +44,9 @@ Arguments for :meth:`sunpy.timeseries.GenericTimeSeries.peek` are now keywords o
 The arguments for :meth:`sunpy.timeseries.GenericTimeSeries.peek` have been changed to be keyword only.
 This means that you must now specify the arguments by name, rather than by position and has been done to make the API more consistent.
 This has been raising a warning since sunpy 4.1 and is now an error.
+
+Deprecation positional arguments in ``plotGenericMap.plot()`` method
+---------------------------------------------------------------------
+
+The arguments for ``plotGenericMap.plot()`` have been changed to being keyword only.
+Pass them as keyword arguments (e.g., ``..., title=True, ...``) instead.
```
Comment (nabobalis) [inline docs/whatsnew/6.0.rst:52]:
```suggestion
Deprecation positional arguments in ``GenericMap.plot()`` method
---------------------------------------------------------------------

The arguments for ``GenericMap.plot()`` have been changed to being keyword only.
Pass them as keyword arguments (e.g., ``..., title=True, ...``) instead.
```

File: changelog/7166.deprecation.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- Deprecations:
+  - Deprecate positional arguments in `plot` method.
+
+    Passing the ``annotate``, ``axes``, ``title``, ``clip_interval`` arguments as positional arguments is deprecated.
+    Pass them as keyword arguments (e.g., ``..., title=True, ...``) instead.
```
Comment (nabobalis) [inline changelog/7166.deprecation.rst:5]:
```suggestion
Deprecated all positional arguments in :meth:`sunpy.map.GenericMap.plot` method.
The ``annotate``, ``axes``, ``title``, ``clip_interval`` arguments should be passed as keyword arguments (e.g., ``..., title=True, ...``) instead.
```

**Comment by nabobalis** [comment]
If you can also update the code from the sunpy main branch, we can see if the tests pass

**Comment by yeppAniket** [comment]
> If you can also update the code from the sunpy main branch, we can see if the tests pass

Like do another pr with all the changes and suggestions combined?

**Comment by wtbarnes** [comment]
> > If you can also update the code from the sunpy main branch, we can see if the tests pass
> 
> Like do another pr with all the changes and suggestions combined?

I believe what Nabil meant was to rebase or merge in the changes from the `main` branch. No need to open a new PR. One advantage syncing your branch with the main branch is that you can take advantage of bug fixes that have been applied elsewhere. 

I've gone ahead and done this for you, but for future reference, you might have a look at this guide for keeping your branch in sync: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch.

**Comment by wtbarnes** [comment]
The test fail is real here and is due to the existing test still passing a positional argument to `plot`: https://github.com/sunpy/sunpy/actions/runs/7893166560/job/21541210647?pr=7421#step:10:5025

I imagine there will be others that do the same though I've not done an exhaustive check of where. I would suggest running the tests in `sunpy/map` locally to more easily find the failures. That will be easier than waiting for the CI to run the full test suite.

**Comment by nabobalis** [comment]
> > If you can also update the code from the sunpy main branch, we can see if the tests pass
> 
> Like do another pr with all the changes and suggestions combined?

Sorry @yeppAniket, I missed this message completely.

**Review by nabobalis — APPROVED** [review]

File: sunpy/util/decorators.py [hunk 3]
```diff
@@ -262,41 +262,24 @@ def __call__(self, func):
         return func
 
 
-def deprecate_positional_args_since(since, keyword_only=False):
+def deprecate_positional_args_since(func=None, *, since):
```
Comment (nabobalis) [inline sunpy/util/decorators.py:265]: I updated this from upstream.

**Comment by nabobalis** [comment]
Thanks for the PR @yeppAniket

**Pull request merged**